pkg/controller/fileobserver/observer_polling.go

```diff
@@ -20,6 +20,12 @@ type pollingObserver struct {
 	files    map[string]string
 
 	reactorsMutex sync.RWMutex
+	hasSynced     bool
+}
+
+// HasSynced indicates that the observer synced all observed files at least once.
+func (o *pollingObserver) HasSynced() bool {
+	return o.hasSynced
```

Comment:
must this be go routine safe?

Reply:
i thought so as well, but aren't booleans safe in go?

Reply:
and I was wrong ;-) the data race in unit test :-)

Reply:
No, they are not.

Reply:
fixed by mutex